shared/actions/provision.tsx

```diff
@@ -318,8 +346,15 @@ class ProvisioningManager {
     response.result({passphrase: password, storeSecret: false})
   }
 
+  displaySecretExchanged = (_params: CustomParam<'keybase.1.provisionUi.DisplaySecretExchanged'>) => {
```

Comment:
this is new. with the built-in waiting handling the code assumes that when we get a callback from the server we're not waiting and when we respond we're waiting. provisioning is the only time when we can sit there and get an incoming that we didn't explicitly ask for which causes our waiting key to go negative. This has been what's messing up the waiting states in this flow sometimes in a very hard to debug way

Reply:
Hitting "enter" with an empty secret phrase is an easy repro to get stuck with a waiting key wrong. It may be good to clear the waiting on `cancel` or something so it's easier to escape.